pkg/tls/self_signed_cert.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package gencert generates self-signed TLS certificates.
+package tls
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"os"
+	"time"
+)
+
+const (
+	caOrganization      = "F5 Inc. CA"
+	certOrganization    = "F5 Inc."
+	certFilePermissions = 0o600
+	keyFilePermissions  = 0o600
+)
+
+type certReq struct {
+	template   *x509.Certificate
+	parent     *x509.Certificate
+	publicKey  *ecdsa.PublicKey
+	privateKey *ecdsa.PrivateKey
+}
+
+func genCert(req *certReq) (*x509.Certificate, []byte) {
+	certBytes, createCertErr := x509.CreateCertificate(
+		rand.Reader,
+		req.template,
+		req.parent,
+		req.publicKey,
+		req.privateKey,
+	)
+
+	if createCertErr != nil {
+		slog.Error("Failed to generate certificate", "error", createCertErr)
+		return &x509.Certificate{}, []byte{}
+	}
+
+	cert, parseCertErr := x509.ParseCertificate(certBytes)
+	if parseCertErr != nil {
+		slog.Error("Failed to parse certificate")
+		return &x509.Certificate{}, []byte{}
+	}
+
+	b := pem.Block{Type: "CERTIFICATE", Bytes: certBytes}
+	certPEM := pem.EncodeToMemory(&b)
+
+	return cert, certPEM
+}
+
```

Comment:
Might be good to add docs to these functions as well like what we did for file_helpers.go https://github.com/nginx/agent/blob/v3/pkg/files/file_helpers.go#L62